### PR TITLE
docs: reatom-react.md spelling and grammar

### PR DESCRIPTION
**Pull request by MunGell**
This PR needs semantic review

### CODE REVIEW

File: docs/reatom-react.md [hunk 1]
```diff
@@ -29,7 +29,7 @@ Connects the atom to the store represented in context and returns the state of t
 const atomValue = useAtom(atom)
 ```
 
-#### Depended value by selector
+#### Dependent value by selector
```
Comment (MunGell) [inline docs/reatom-react.md]: Not sure what the original heading meant, please confirm

Comment (MunGell) [reply docs/reatom-react.md]: @belozer "Dynamic atom, which will be re-created on dependency change" ?

Comment (belozer) [reply docs/reatom-react.md]: @MunGell hard to understand

Comment (MunGell) [reply docs/reatom-react.md]: Maybe with description like that:

#### Dynamic atom

Dynamic atoms are re-created on dependency change (passed as the second argument). This technique is useful when "<something>".

<code>

Comment (belozer) [reply docs/reatom-react.md]: Maybe like: **With selector (render on value change)**

Comment (MunGell) [reply docs/reatom-react.md]: I've updated the wording there, please check

**Pull request merged**